services/core/SQLHistorian/sqlhistorian/historian.py

```diff
@@ -370,6 +339,20 @@ def historian_setup(self):
         thread_name = threading.currentThread().getName()
         _log.debug("historian_setup on Thread: {}".format(thread_name))
 
+        database_type = self.config['connection']['type']
+        self.tables_def, table_names = self.parse_table_def(self.config)
+        db_functs_class = sqlutils.get_dbfuncts_class(database_type)
+        self.reader = db_functs_class(self.config['connection']['params'],
+                                      table_names)
+        self.writer = db_functs_class(self.config['connection']['params'],
+                                      table_names)
+        self.reader.setup_historian_tables()
```

Comment:
I know they are the same class but it seems wrong to have a reader setting up tables